### PR TITLE
Admin Router: Documentation on generic tests

### PR DESCRIPTION
**Pull request by vespian**
## High Level Description

This PR adds tests documentation.

## Related issues:
https://jira.mesosphere.com/browse/DCOS_OSS-1012 Admin Router: Quick-start guide for writing tests

## Related PRs:
AR-NEXT: https://github.com/dcos/dcos/pull/1802

### CODE REVIEW

**Review by gpaul — APPROVED** [review]
Super useful docs, I've wondered about this stuff in the past.

File: packages/adminrouter/extra/src/README.md [hunk 1]
```diff
@@ -716,6 +716,222 @@ done by `test-harness/modules/mocker/jwt.py` module, which together with
...
+  list of header tuples contains or does not contain given header.
+* `GuardedSubprocess` - launches given AR instance and makes sure that it is
+  terminated at the end of the context.
+* `iam_denies_all_requests` - makes sure that the IAM denies all PolicyQuery
+  requests within the context.
+* `ping_mesos_agent` - tests if agent with a given ID
+  (`mocker.endpoints.mesos.AGENT1_ID` by default) can be reached.
+* `repo_is_ee` - detects repository flavour - whether repository is Open DC/OS
+  or Enterprise DC/OS.
+* `assert_iam_queried_for_uid` - asserts that within the context, IAM has been
```
Comment (jgehrcke) [inline packages/adminrouter/extra/src/README.md:762]: In the big PR I asked a question about this one, and was hoping for clarification.

Did you consider my comments in the big PR?

**Review by jgehrcke — CHANGES_REQUESTED** [review]

File: packages/adminrouter/extra/src/README.md (hunk 1 above)
Comment (jgehrcke) [reply packages/adminrouter/extra/src/README.md:762]: Nevermind, sorry for asking this question :) Found https://github.com/dcos/dcos/pull/1702#discussion_r131408670.

**Review by jgehrcke — APPROVED** [review]
Thank you! (!)
:rocket:

**Pull request merged**